src/test/kotlin/pl/newbies/auth/GithubAuthTest.kt

```diff
@@ -0,0 +1,79 @@
+package pl.newbies.auth
+
+import io.ktor.client.plugins.ClientRequestException
+import io.ktor.client.request.forms.submitForm
+import io.ktor.client.request.get
+import io.ktor.client.statement.bodyAsText
+import io.ktor.http.HttpStatusCode
+import io.ktor.http.Parameters
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.Nested
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertThrows
+import pl.newbies.util.IntegrationTest
+import pl.newbies.util.TestData
+import pl.newbies.util.httpClient
+import pl.newbies.util.loginAs
+
+class GithubAuthTest : IntegrationTest() {
+
+    @Nested
+    inner class Login {
+        @Test
+        fun `should redirect when login called`() = withAres {
+            // when
+            val response = httpClient.get("/oauth/login/github")
+
+            // then
+            assertEquals(HttpStatusCode.OK, response.status)
+            assertEquals("ok", response.bodyAsText())
+        }
+    }
+
+    @Nested
+    inner class Callback {
+        @Test
+        fun `should create account when called with new user`() = withAres {
+            // given
+            clearTable("Users")
+            val user = TestData.testUser1
+
+            // when
+            val response = loginAs(user)
+
+            // then
+            assertEquals(user.name, response.username)
+        }
+
+        @Test
+        fun `should return existing account when called with existing github user`() = withAres {
+            // given
+            clearTable("Users")
```

Comment:
i can't determine if user already exists bc this test looks exactly the same as one above

Reply:
Theoretically user should be created when first `loginAs(...)` is called, this test call loginAs twice to check if the second call actually returns user, maybe I should do some "createDate" comparison to determine if it is actually the previous user

Reply:
my bad
didn't see second one

Reply:
I've added assertions to check if createDate is the same after second login, that should clarify that user data is actually persisted 😄